java/arcs/core/entity/CollectionHandle.kt

```diff
@@ -45,7 +45,9 @@ class CollectionHandle<T : Storable, R : Referencable>(
   private val storageAdapter = config.storageAdapter
 
   init {
-    check(spec.containerType == HandleContainerType.Collection)
+    check(spec.containerType == HandleContainerType.Collection) {
+      "HandleContainerType of ${name} does not match Spec containerType."
```

Comment:
You could be even clearer, and say something like: `"Collection container type required for CollectionHandle $name, but got ${spec.containerType}."`